### ISSUE TITLE
/etc Makefile Does Not Respect --prefix

### ISSUE THREAD

**Issue by prikhi** [IgnorantGuru/udevil#53]
I'm trying to create a NixOS package for udevil.

Nix works by installing everything to /nix/store then symlinking the "currently installed" software to the root directories like /bin, /etc, /share.

So a package should create directories like `/nix/store/buildhash-udevil-version/etc` then Nix will symlink the files and folders in the store directory to /etc.

In order to do this, Nix adds the `--prefix=/nix/store/buildhash-udevil-version/` option when running `./configure`.

Unfortunately, during the `make install` step, udevil attempts to make the `/etc/udevil` directory, even when a prefix is  specified. This occurs in the Makefile for the `/etc` directory (the `install-data-hook` step).

Here is the Nix expression I am using to build the package:
```
{ stdenv, fetchurl, glib, udev, fakeroot, intltool, pkgconfig }:                
                                                                                
stdenv.mkDerivation rec {                                                       
  version = "0.4.3";                                                            
  name = "udevil-${version}";                                                   
                                                                                
  src = fetchurl {                                                              
    url = "http://github.com/IgnorantGuru/udevil/archive/${version}.tar.gz";  
    sha256 = "02b2pk7z3ms8j6718jdprsi564pyfn7p03amc30q4wgqp922d7f4";            
  };                                                                            
                                                                                
  buildInputs = [ glib udev fakeroot intltool pkgconfig ];                    
                                                                                                                                                                                
  preferLocalBuild = true;                                                    
                                                                              
  meta = with stdenv.lib; {                                                   
    description = "Mount without password";                                   
    longDescription = ''                                                      
      Mounts and unmounts removable devices and networks without a password   
      (set suid), shows device info, monitors device changes. Emulates mount's
      and udisks's command line usage and udisks v1's output. Includes the    
      devmon automounting daemon.                                             
                                                                              
      Run 'udevil --help' for usage.  See /etc/udevil/udevil.conf for config. 
                                                                              
      http://ignorantguru.github.com/udevil/udevil--help.html                 
      http://ignorantguru.github.com/udevil/udevil.conf.txt                   
    '';                                                                       
    homepage = https://github.com/IgnorantGuru/udevil;                        
    downloadPage = https://github.com/IgnorantGuru/udevil/releases;           
    license = licenses.gpl3Plus;                                              
    maintainers = [ maintainers.prikhi ];                                     
    platforms = platforms.linux;                                              
  };                                                                          
} 
```

And the output from configure/make:
```
installing `udevil-0.4.3'
these derivations will be built:
  /nix/store/idfrwzn82sjwbs1sw8ppb5gzlhhc307r-udevil-0.4.3.drv
building path(s) `/nix/store/536pny4cdwf7jzx0xckcc0qc95g0wicc-udevil-0.4.3'
building /nix/store/536pny4cdwf7jzx0xckcc0qc95g0wicc-udevil-0.4.3
unpacking sources
unpacking source archive /nix/store/jfh7cnyz7zc1xmj6pjhb8r8xmzkrar4r-0.4.3.tar.gz
source root is udevil-0.4.3
patching sources
configuring
fixing libtool script ./ltmain.sh
configure flags: --disable-static --disable-dependency-tracking --prefix=/nix/store/536pny4cdwf7jzx0xckcc0qc95g0wicc-udevil-0.4.3  
checking for a BSD-compatible install... /nix/store/581vm17nkn6g1hcq8gdvnmhkvn4msbsy-coreutils-8.23/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /nix/store/581vm17nkn6g1hcq8gdvnmhkvn4msbsy-coreutils-8.23/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether NLS is requested... yes
checking for style of include used by make... GNU
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking dependency style of gcc... none
checking for intltool >= 0.21... 0.50.2 found
checking for intltool-update... /nix/store/k6ahpblz2zm8252a990rp3wqxjx7vs85-intltool-0.50.2/bin/intltool-update
checking for intltool-merge... /nix/store/k6ahpblz2zm8252a990rp3wqxjx7vs85-intltool-0.50.2/bin/intltool-merge
checking for intltool-extract... /nix/store/k6ahpblz2zm8252a990rp3wqxjx7vs85-intltool-0.50.2/bin/intltool-extract
checking for xgettext... /nix/store/xcr79dff95kgd2bca4pnrhnvlih855md-gettext-0.18.2/bin/xgettext
checking for msgmerge... /nix/store/xcr79dff95kgd2bca4pnrhnvlih855md-gettext-0.18.2/bin/msgmerge
checking for msgfmt... /nix/store/xcr79dff95kgd2bca4pnrhnvlih855md-gettext-0.18.2/bin/msgfmt
checking for gmsgfmt... /nix/store/xcr79dff95kgd2bca4pnrhnvlih855md-gettext-0.18.2/bin/msgfmt
checking for perl... /nix/store/pxmj3a5lm13v9zqdclr18874r28iq606-perl-5.16.3/bin/perl
checking for perl >= 5.8.1... 5.16.3
checking for XML::Parser... ok
checking whether to enable maintainer-specific portions of Makefiles... no
checking for library containing strerror... none required
checking how to run the C preprocessor... gcc -E
checking for grep that handles long lines and -e... /nix/store/a6jx7dzh299ap2rh3di57gh48jiy18zg-gnugrep-2.20/bin/grep
checking for egrep... /nix/store/a6jx7dzh299ap2rh3di57gh48jiy18zg-gnugrep-2.20/bin/grep -E
checking for ANSI C header files... yes
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking how to print strings... printf
checking for a sed that does not truncate output... /nix/store/kd8ab7h91rb7gwmmk4fml8778l8izd04-gnused-4.2.2/bin/sed
checking for fgrep... /nix/store/a6jx7dzh299ap2rh3di57gh48jiy18zg-gnugrep-2.20/bin/grep -F
checking for ld used by gcc... /nix/store/q547sj1apn6m7cqg6qlhlfw6v5viwwzi-gcc-wrapper-4.8.3/bin/ld
checking if the linker (/nix/store/q547sj1apn6m7cqg6qlhlfw6v5viwwzi-gcc-wrapper-4.8.3/bin/ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /nix/store/vi0wcqxwxqrrnmpm0mxbq84rgw7s16xw-binutils-2.23.1/bin/nm -B
checking the name lister (/nix/store/vi0wcqxwxqrrnmpm0mxbq84rgw7s16xw-binutils-2.23.1/bin/nm -B) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking whether the shell understands some XSI constructs... yes
checking whether the shell understands "+="... yes
checking how to convert x86_64-unknown-linux-gnu file names to x86_64-unknown-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-unknown-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /nix/store/q547sj1apn6m7cqg6qlhlfw6v5viwwzi-gcc-wrapper-4.8.3/bin/ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for ar... ar
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse /nix/store/vi0wcqxwxqrrnmpm0mxbq84rgw7s16xw-binutils-2.23.1/bin/nm -B output from gcc object... ok
checking for sysroot... no
./configure: line 7272: /usr/bin/file: No such file or directory
checking for mt... no
checking if : is a manifest tool... no
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC -DPIC
checking if gcc PIC flag -fPIC -DPIC works... yes
checking if gcc static flag -static works... yes
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (/nix/store/q547sj1apn6m7cqg6qlhlfw6v5viwwzi-gcc-wrapper-4.8.3/bin/ld) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no
checking whether /nix/store/q547sj1apn6m7cqg6qlhlfw6v5viwwzi-gcc-wrapper-4.8.3/bin/ld accepts --as-needed... yes
checking whether /nix/store/q547sj1apn6m7cqg6qlhlfw6v5viwwzi-gcc-wrapper-4.8.3/bin/ld accepts -O1... yes
checking whether /nix/store/q547sj1apn6m7cqg6qlhlfw6v5viwwzi-gcc-wrapper-4.8.3/bin/ld accepts -Bsymbolic-functions... yes
checking for pkg-config... /nix/store/5kjxj474bj02hg3w6vq6ps9yg7y5g8rk-pkg-config-0.28/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for GLIB... yes
checking for LIBUDEV... yes
checking locale.h usability... yes
checking locale.h presence... yes
checking for locale.h... yes
checking for LC_MESSAGES... yes
checking libintl.h usability... yes
checking libintl.h presence... yes
checking for libintl.h... yes
checking for ngettext in libc... yes
checking for dgettext in libc... yes
checking for bind_textdomain_codeset... yes
checking for msgfmt... (cached) /nix/store/xcr79dff95kgd2bca4pnrhnvlih855md-gettext-0.18.2/bin/msgfmt
checking for dcgettext... yes
checking if msgfmt accepts -c... yes
checking for gmsgfmt... (cached) /nix/store/xcr79dff95kgd2bca4pnrhnvlih855md-gettext-0.18.2/bin/msgfmt
checking for xgettext... (cached) /nix/store/xcr79dff95kgd2bca4pnrhnvlih855md-gettext-0.18.2/bin/xgettext
checking for catalogs to be installed...  ru
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating Makefile
config.status: creating po/Makefile.in
config.status: creating man/Makefile
config.status: creating etc/Makefile
config.status: creating src/Makefile
config.status: creating config.h
config.status: executing depfiles commands
config.status: executing libtool commands
config.status: executing default-1 commands
config.status: executing po/stamp-it commands

udevil....................................... : Version 0.4.3

Linux device support......................... : libudev
mount program................................ : /bin/mount
umount program............................... : /bin/umount
losetup program.............................. : /sbin/losetup
setfacl program.............................. : /usr/bin/setfacl
install devmon systemd service............... : yes

Installation Prefix.......................... : /nix/store/536pny4cdwf7jzx0xckcc0qc95g0wicc-udevil-0.4.3
      The binary will be installed as /nix/store/536pny4cdwf7jzx0xckcc0qc95g0wicc-udevil-0.4.3/bin/udevil

Homepage: http://ignorantguru.github.com/udevil/

building
make flags: SHELL=/nix/store/kly624h55gf91frdhn8fjzi6xricgkzk-bash-4.2-p53/bin/bash    
building config.h
building all
 make  all-recursive
make[1]: Entering directory `/tmp/nix-build-udevil-0.4.3.drv-0/udevil-0.4.3'
building all-recursive
Making all in etc
make[2]: Entering directory `/tmp/nix-build-udevil-0.4.3.drv-0/udevil-0.4.3/etc'
make[2]: Nothing to be done for `all'.
make[2]: Leaving directory `/tmp/nix-build-udevil-0.4.3.drv-0/udevil-0.4.3/etc'
Making all in po
make[2]: Entering directory `/tmp/nix-build-udevil-0.4.3.drv-0/udevil-0.4.3/po'
building de.gmo
 file=`echo de | sed 's,.*/,,'`.gmo \
  && rm -f $file && /nix/store/xcr79dff95kgd2bca4pnrhnvlih855md-gettext-0.18.2/bin/msgfmt -o $file de.po
building es.gmo
 file=`echo es | sed 's,.*/,,'`.gmo \
  && rm -f $file && /nix/store/xcr79dff95kgd2bca4pnrhnvlih855md-gettext-0.18.2/bin/msgfmt -o $file es.po
building fr.gmo
 file=`echo fr | sed 's,.*/,,'`.gmo \
  && rm -f $file && /nix/store/xcr79dff95kgd2bca4pnrhnvlih855md-gettext-0.18.2/bin/msgfmt -o $file fr.po
building it.gmo
 file=`echo it | sed 's,.*/,,'`.gmo \
  && rm -f $file && /nix/store/xcr79dff95kgd2bca4pnrhnvlih855md-gettext-0.18.2/bin/msgfmt -o $file it.po
building pl.gmo
 file=`echo pl | sed 's,.*/,,'`.gmo \
  && rm -f $file && /nix/store/xcr79dff95kgd2bca4pnrhnvlih855md-gettext-0.18.2/bin/msgfmt -o $file pl.po
building pt_BR.gmo
 file=`echo pt_BR | sed 's,.*/,,'`.gmo \
  && rm -f $file && /nix/store/xcr79dff95kgd2bca4pnrhnvlih855md-gettext-0.18.2/bin/msgfmt -o $file pt_BR.po
building ru.gmo
 file=`echo ru | sed 's,.*/,,'`.gmo \
  && rm -f $file && /nix/store/xcr79dff95kgd2bca4pnrhnvlih855md-gettext-0.18.2/bin/msgfmt -o $file ru.po
building sv.gmo
 file=`echo sv | sed 's,.*/,,'`.gmo \
  && rm -f $file && /nix/store/xcr79dff95kgd2bca4pnrhnvlih855md-gettext-0.18.2/bin/msgfmt -o $file sv.po
make[2]: Leaving directory `/tmp/nix-build-udevil-0.4.3.drv-0/udevil-0.4.3/po'
Making all in src
make[2]: Entering directory `/tmp/nix-build-udevil-0.4.3.drv-0/udevil-0.4.3/src'
building udevil-udevil.o
 gcc -DHAVE_CONFIG_H -I. -I.. -DPACKAGE_LOCALE_DIR=\""/nix/store/536pny4cdwf7jzx0xckcc0qc95g0wicc-udevil-0.4.3/share/locale"\"  -fstrict-aliasing -fmessage-length=0 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_REENTRANT -D_FILE_OFFSET_BITS=64 -I/nix/store/2vfjv98il57s88qaqasjkh1jygbzi4xm-glib-2.42.1/include/glib-2.0 -I/nix/store/2vfjv98il57s88qaqasjkh1jygbzi4xm-glib-2.42.1/lib/glib-2.0/include -I/nix/store/nxzp7kmxkyp55n8bx0b6lw40w4vcpnwq-systemd-217/include -Werror-implicit-function-declaration -g -O2 -c -o udevil-udevil.o `test -f 'udevil.c' || echo './'`udevil.c
building udevil-device-info.o
 gcc -DHAVE_CONFIG_H -I. -I.. -DPACKAGE_LOCALE_DIR=\""/nix/store/536pny4cdwf7jzx0xckcc0qc95g0wicc-udevil-0.4.3/share/locale"\"  -fstrict-aliasing -fmessage-length=0 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_REENTRANT -D_FILE_OFFSET_BITS=64 -I/nix/store/2vfjv98il57s88qaqasjkh1jygbzi4xm-glib-2.42.1/include/glib-2.0 -I/nix/store/2vfjv98il57s88qaqasjkh1jygbzi4xm-glib-2.42.1/lib/glib-2.0/include -I/nix/store/nxzp7kmxkyp55n8bx0b6lw40w4vcpnwq-systemd-217/include -Werror-implicit-function-declaration -g -O2 -c -o udevil-device-info.o `test -f 'device-info.c' || echo './'`device-info.c
building udevil-canonicalize.o
 gcc -DHAVE_CONFIG_H -I. -I.. -DPACKAGE_LOCALE_DIR=\""/nix/store/536pny4cdwf7jzx0xckcc0qc95g0wicc-udevil-0.4.3/share/locale"\"  -fstrict-aliasing -fmessage-length=0 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_REENTRANT -D_FILE_OFFSET_BITS=64 -I/nix/store/2vfjv98il57s88qaqasjkh1jygbzi4xm-glib-2.42.1/include/glib-2.0 -I/nix/store/2vfjv98il57s88qaqasjkh1jygbzi4xm-glib-2.42.1/lib/glib-2.0/include -I/nix/store/nxzp7kmxkyp55n8bx0b6lw40w4vcpnwq-systemd-217/include -Werror-implicit-function-declaration -g -O2 -c -o udevil-canonicalize.o `test -f 'canonicalize.c' || echo './'`canonicalize.c
building udevil
 /nix/store/kly624h55gf91frdhn8fjzi6xricgkzk-bash-4.2-p53/bin/bash ../libtool  --tag=CC   --mode=link gcc -I/nix/store/2vfjv98il57s88qaqasjkh1jygbzi4xm-glib-2.42.1/include/glib-2.0 -I/nix/store/2vfjv98il57s88qaqasjkh1jygbzi4xm-glib-2.42.1/lib/glib-2.0/include -I/nix/store/nxzp7kmxkyp55n8bx0b6lw40w4vcpnwq-systemd-217/include -Werror-implicit-function-declaration -g -O2  -Wl,--as-needed -Wl,-O1 -Wl,-Bsymbolic-functions -o udevil udevil-udevil.o udevil-device-info.o udevil-canonicalize.o  -L/nix/store/2vfjv98il57s88qaqasjkh1jygbzi4xm-glib-2.42.1/lib -lglib-2.0 -L/nix/store/nxzp7kmxkyp55n8bx0b6lw40w4vcpnwq-systemd-217/lib -ludev  
libtool: link: gcc -I/nix/store/2vfjv98il57s88qaqasjkh1jygbzi4xm-glib-2.42.1/include/glib-2.0 -I/nix/store/2vfjv98il57s88qaqasjkh1jygbzi4xm-glib-2.42.1/lib/glib-2.0/include -I/nix/store/nxzp7kmxkyp55n8bx0b6lw40w4vcpnwq-systemd-217/include -Werror-implicit-function-declaration -g -O2 -Wl,--as-needed -Wl,-O1 -Wl,-Bsymbolic-functions -o udevil udevil-udevil.o udevil-device-info.o udevil-canonicalize.o  -L/nix/store/2vfjv98il57s88qaqasjkh1jygbzi4xm-glib-2.42.1/lib /nix/store/2vfjv98il57s88qaqasjkh1jygbzi4xm-glib-2.42.1/lib/libglib-2.0.so -L/nix/store/nxzp7kmxkyp55n8bx0b6lw40w4vcpnwq-systemd-217/lib /nix/store/nxzp7kmxkyp55n8bx0b6lw40w4vcpnwq-systemd-217/lib/libudev.so -pthread -Wl,-rpath -Wl,/nix/store/2vfjv98il57s88qaqasjkh1jygbzi4xm-glib-2.42.1/lib -Wl,-rpath -Wl,/nix/store/nxzp7kmxkyp55n8bx0b6lw40w4vcpnwq-systemd-217/lib -Wl,-rpath -Wl,/nix/store/2vfjv98il57s88qaqasjkh1jygbzi4xm-glib-2.42.1/lib -Wl,-rpath -Wl,/nix/store/nxzp7kmxkyp55n8bx0b6lw40w4vcpnwq-systemd-217/lib
make[2]: Leaving directory `/tmp/nix-build-udevil-0.4.3.drv-0/udevil-0.4.3/src'
make[2]: Entering directory `/tmp/nix-build-udevil-0.4.3.drv-0/udevil-0.4.3'
building config.h
make[2]: Leaving directory `/tmp/nix-build-udevil-0.4.3.drv-0/udevil-0.4.3'
make[1]: Leaving directory `/tmp/nix-build-udevil-0.4.3.drv-0/udevil-0.4.3'
installing
install flags: install SHELL=/nix/store/kly624h55gf91frdhn8fjzi6xricgkzk-bash-4.2-p53/bin/bash    gsettingsschemadir=/nix/store/536pny4cdwf7jzx0xckcc0qc95g0wicc-udevil-0.4.3/share/gsettings-schemas/udevil-0.4.3/glib-2.0/schemas/
building install-recursive
Making install in etc
make[1]: Entering directory `/tmp/nix-build-udevil-0.4.3.drv-0/udevil-0.4.3/etc'
building install-am
make[2]: Entering directory `/tmp/nix-build-udevil-0.4.3.drv-0/udevil-0.4.3/etc'
make[2]: Nothing to be done for `install-exec-am'.
building install-data-am
 make  install-data-hook
make[3]: Entering directory `/tmp/nix-build-udevil-0.4.3.drv-0/udevil-0.4.3/etc'
building install-data-hook
 test -d /etc/udevil || mkdir -p -- /etc/udevil
mkdir: cannot create directory '/etc/udevil': Permission denied
make[3]: *** [install-data-hook] Error 1
make[3]: Leaving directory `/tmp/nix-build-udevil-0.4.3.drv-0/udevil-0.4.3/etc'
make[2]: *** [install-data-am] Error 2
make[2]: Leaving directory `/tmp/nix-build-udevil-0.4.3.drv-0/udevil-0.4.3/etc'
make[1]: *** [install-am] Error 2
make[1]: Leaving directory `/tmp/nix-build-udevil-0.4.3.drv-0/udevil-0.4.3/etc'
make: *** [install-recursive] Error 1
builder for `/nix/store/idfrwzn82sjwbs1sw8ppb5gzlhhc307r-udevil-0.4.3.drv' failed with exit code 2
error: build of `/nix/store/idfrwzn82sjwbs1sw8ppb5gzlhhc307r-udevil-0.4.3.drv' failed
```

You can see the correct path is set during the configure step:
```
Installation Prefix.......................... : /nix/store/536pny4cdwf7jzx0xckcc0qc95g0wicc-udevil-0.4.3
      The binary will be installed as /nix/store/536pny4cdwf7jzx0xckcc0qc95g0wicc-udevil-0.4.3/bin/udevil
```

Ideally the Makefile in the `/etc` sub-directory would also use this path, but for some reason this is not happening.

**Comment by IgnorantGuru** [comment]
Thanks this is already being addressed in pull request #31

**Comment by IgnorantGuru** [comment]
This has been fixed in the next branch - please see BUILD NEXT instructions in README to test that branch.

Option --sysconfdir must be specified to set a dir other than /etc.  For example, specify locations like this:

    ./configure --prefix=/usr --sysconfdir=/etc

If using systemd (for starting devmon), you must also modify the contents of *etc/systemd/devmon@.service* in your build patches, per the updated README in that dir.

Thanks to @OmegaPhil for doing most of the work on this (#31).
